### PR TITLE
Support distributed load files to alluxio by a include worker set

### LINKED ISSUE #13221: Support distributedload into specific worker list.
**Is your feature request related to a problem? Please describe.**

Now, Alluxio shell command distributedload can distributed load a directory into alluxio cluster, in our scenario, we want to distributedload the files into specific alluxio workers.

**Describe the solution you'd like**

Add an option workerList to limit the worker which will load files into alluxio tired store.

**Urgency**
Nornal

**Additional context**
NO

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #13221

Usage:

```
bin/alluxio fs distributedLoad / --worker-set host1,host2,host3
```

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 1]
```diff
@@ -109,7 +125,14 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI path = new AlluxioURI(args[0]);
     int replication = FileSystemShellUtils.getIntArg(cl, REPLICATION_OPTION, DEFAULT_REPLICATION);
-    distributedLoad(path, replication);
+    Set<String> workerSet = Collections.EMPTY_SET;
+    if (cl.hasOption(WORKER_SET_OPTION.getLongOpt())) {
+      String argOption = cl.getOptionValue(WORKER_SET_OPTION.getLongOpt());
+      String[] workerArray = StringUtils.split(argOption, ",");
+      workerSet = new HashSet<>();
+      Collections.addAll(workerSet, workerArray);
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:133]: Is it possible that this adds empty strings, or can hostnames have whitespace around them?

File: job/common/src/main/java/alluxio/job/plan/load/LoadConfig.java [hunk 2]
```diff
@@ -32,15 +33,19 @@
...
    * @param replication the number of workers to store each block on, defaults to 1
+   * @param workerSet the worker set
    */
   public LoadConfig(@JsonProperty("filePath") String filePath,
-      @JsonProperty("replication") Integer replication) {
+      @JsonProperty("replication") Integer replication,
+      @JsonProperty("workerSet") Set<String> workerSet) {
     mFilePath = Preconditions.checkNotNull(filePath, "The file path cannot be null");
     mReplication = replication == null ? 1 : replication;
+    mWorkerSet = workerSet;
```
Comment (gpang) [inline job/common/src/main/java/alluxio/job/plan/load/LoadConfig.java:48]: should this make a copy of the set, instead of referring to the input set?

File: job/common/src/test/java/alluxio/job/plan/load/LoadConfigTest.java [hunk 3]
```diff
@@ -15,13 +15,15 @@
...
+import java.util.Collections;
+
 /**
  * Tests {@link LoadConfig}.
  */
 public final class LoadConfigTest {
   @Test
   public void jsonTest() throws Exception {
-    LoadConfig config = new LoadConfig("/path/to/load", 3);
+    LoadConfig config = new LoadConfig("/path/to/load", 3, Collections.EMPTY_SET);
```
Comment (gpang) [inline job/common/src/test/java/alluxio/job/plan/load/LoadConfigTest.java:26]: Can you add a test case where the set is not empty?

File: job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java [hunk 4]
```diff
@@ -73,7 +73,11 @@ public LoadDefinition() {
     List<BlockWorkerInfo> workers = new ArrayList<>();
     for (BlockWorkerInfo worker : context.getFsContext().getCachedWorkers()) {
       if (jobWorkersByAddress.containsKey(worker.getNetAddress().getHost())) {
-        workers.add(worker);
+        if (config.getWorkerSet() == null
+            || config.getWorkerSet().isEmpty()
+            || config.getWorkerSet().contains(worker.getNetAddress().getHost())) {
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java:78]: Would case sensitivity cause problems?

**Review by gpang — COMMENTED** [review]
@maobaolong Thanks! I left a few comments.

File: job/common/src/main/java/alluxio/job/plan/load/LoadConfig.java (hunk 2 above)
Comment (apc999) [reply job/common/src/main/java/alluxio/job/plan/load/LoadConfig.java:48]: Let's avoid the possibility to have `mWorkerSet` `null`?
```suggestion
    mWorkerSet = workerSet == null ? Collections.EMPTY_SET : new HashSet(workerSet) ;
```

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 5]
```diff
@@ -65,6 +69,15 @@
                   + "The default upper limit is "
                   + AbstractDistributedJobCommand.DEFAULT_ACTIVE_JOBS)
           .build();
+  private static final Option WORKER_SET_OPTION =
+      Option.builder()
+          .longOpt("worker-set")
```
Comment (apc999) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:74]: how about `--host`? 

```suggestion
          .longOpt("host")
```

File: job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java (hunk 4 above)
Comment (apc999) [reply job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java:78]: also ip vs hostname?

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 6]
```diff
@@ -77,6 +82,15 @@
           .argName("index file")
           .desc("Name of the index file that lists all files to be loaded")
           .build();
+  private static final Option WORKER_SET_OPTION =
```
Comment (apc999) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:85]:
```suggestion
  private static final Option HOSTS_OPTION =
```

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 7]
```diff
@@ -77,6 +82,15 @@
...
+      Option.builder()
+          .longOpt("host")
```
Comment (apc999) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:87]:
```suggestion
          .longOpt("hosts")
```

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 8]
```diff
@@ -77,6 +82,15 @@
...
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(1)
+          .argName("host")
+          .desc("Worker host list separated by comma")
```
Comment (apc999) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:92]:
```suggestion
          .desc("A list of worker hosts separated by comma")
```

**Review by apc999 — APPROVED** [review]
minor comments left, I will merge your PR after they are addressed.
Thanks!

**Comment by maobaolong** [comment]
@apc999 Thanks for your review and suggestion, the comments are addressed. PTAL

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 8 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: done

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 7 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: done

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 6 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: done

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please
